client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/CoapChannel.java

```diff
@@ -0,0 +1,265 @@
+    if (demultiplexer != null) {
+      this.demultiplexer = demultiplexer;
+    }
+  }
+
+  @Override
+  public void setMultiplexer(KaaDataMultiplexer multiplexer) {
+    if (multiplexer != null) {
+      this.multiplexer = multiplexer;
+    }
+  }
+
+  @Override
+  public void setServer(TransportConnectionInfo server) {
+
+    LOG.info("Setting server [{}] for channel [{}]", server, getId());
+    if (server == null) {
+      LOG.warn("Server is null for Channel [{}].", getId());
+      return;
+    }
+
+    IpTransportInfo oldServer = currentServer;
+    this.currentServer = new IpTransportInfo(server);
+    //this.encDec = new MessageEncoderDecoder(state.getPrivateKey(), state.getPublicKey(), currentServer.getPublicKey());
+    //uncompelete!
+
+  }
+
+  @Override
+  public TransportConnectionInfo getServer() {
+    return currentServer;
+
+  }
+
+  @Override
+  public void setConnectivityChecker(ConnectivityChecker checker) {
+    //To change body of implemented methods use File | Settings | File Templates.
+  }
+
+  @Override
+  public Map<TransportType, ChannelDirection> getSupportedTransportTypes() {
+    return SUPPORTED_TYPES;
+  }
+
+  @Override
+  public void shutdown() {
+  }
+
+  @Override
+  public void pause() {
+  }
+
+  @Override
+  public void resume() {
+  }
+
+  /**
+   *
+   * @param myUri  uri of source
+   * @param method  CoAP method like POST,GET,...
+   * @param payload Message Payload
+   * @param msgType Confirmable or Non confirmable
+   * @return  server response
+   */
+  public CoapResponse sendData(String myUri, String method, String payload, String msgType) {
+    LOG.info("this is SendData function!");
+    URI uri = null;
+
+    if (myUri != null) {
+      // input URI from command line arguments
+      try {
+        uri = new URI(myUri);
+      } catch (URISyntaxException exception) {
+        LOG.info("Invalid URI: " + exception.getMessage());
+        // System.exit(-1);
+      }
+
+      CoapClient client = new CoapClient(uri);
+      LOG.info(" new coap client is registerd");
+
+      CoapResponse response;
+
+      if (msgType == "NON") {
+        client.useNONs();
+      } else {
+        client.useCONs();
+      }
+
+      if (method == "post") {
+        response = client.post(payload, MediaTypeRegistry.TEXT_PLAIN);
+      } else {
+        response = client.get();
+      }
+
+      if (response != null) {
+
+        LOG.info(String.valueOf(response.getCode()));
+        LOG.info(String.valueOf(response.getOptions()));
+        LOG.info(response.getResponseText());
+
+        LOG.info("\nADVANCED\n");
+        // access advanced API with access to more details through .advanced()
+        LOG.info(Utils.prettyPrint(response));
+
+      } else {
+        LOG.info("No response received.");
+      }
+
+      return response;
+    } else {
+      // display help
+      LOG.info("Californium (Cf) GET Client");
+      LOG.info("(c) 2014, Institute for Pervasive Computing, ETH Zurich");
```

Comment:
Do you own copyrights for the code in question?

[Californium GET Client](https://github.com/eclipse/californium/blob/master/demo-apps/cf-helloworld-client/src/main/java/org/eclipse/californium/examples/GETClient.java) is licensed under Eclipse Public License, which is not mixable with Apache 2.0.